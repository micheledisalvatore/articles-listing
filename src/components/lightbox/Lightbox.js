import React, { Component } from 'react';
import { bindActionCreators } from 'redux';

import { hideArticle } from '../../actions/articles';

import Article from '../article';

import { Wrapper, Background } from './Lightbox.styled';

export class Lightbox extends Component {
  constructor(props){
    super(props);
    this.handleEscKey = this.handleEscKey.bind(this);
  }

  handleEscKey(event){
    const { hideArticleAction } = this.props;

    if(event.keyCode === 27) {
      hideArticleAction();
    }
  }

  componentDidMount(){
    document.addEventListener("keydown", this.handleEscKey, false);
  }

  componentWillUnmount(){
    document.removeEventListener("keydown", this.handleEscKey, false);
  }

  render() {
    const { hideArticleAction, article: {artwork, content, title }} = this.props;

    return (
      <Wrapper>
        <Background onClick={hideArticleAction} />
        <Article artwork={artwork} content={content} title={title} full />
      </Wrapper>
    )
  }
}

export const mapStateToProps = ({ articles: { list, selected } }) => ({
  article: list[selected],
});

export const mapDispatchToProps = (dispatch) => ({
  hideArticleAction: bindActionCreators(hideArticle, dispatch),
})
