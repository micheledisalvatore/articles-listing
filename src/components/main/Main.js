import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { isNil } from 'ramda';

import Grid from '../grid';
import Lightbox from '../lightbox';

export const Main = ({ selectedArticle }) => (
  <Fragment>
    <Grid />
    { !isNil(selectedArticle) && ( <Lightbox /> )}
  </Fragment>
);

Lightbox.propTypes = {
  selectedArticle: PropTypes.number,
}

Lightbox.defaultProps = {
  selectedArticle: null,
}

export const mapStateToProps = ({ articles: { selected }}) => ({
  selectedArticle: selected,
});
