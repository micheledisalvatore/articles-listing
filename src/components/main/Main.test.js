/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';

import Lightbox from '../lightbox';

import { Main, mapStateToProps } from './Main';

describe('Given a Main component', () => {
  let component;
  let wrapper;

  describe('when it is rendered', () => {
    beforeEach(() => {
      component = <Main />;
      wrapper = shallow(component);
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should NOT show the Lightbox', () => {
      expect(wrapper.find(Lightbox)).not.toExist();
    })

    describe('and an article is selected', () => {
      beforeEach(() => {
        wrapper.setProps({
          selectedArticle: 1,
        })
      });

      it('should show the Lightbox', () => {
        expect(wrapper.find(Lightbox)).toExist();
      })
    })
  });
});

describe('Given a mapStateToProps function', () => {
  describe('when it is called', () => {
    it('should return an object with the passed values', () => {
      expect(mapStateToProps({
        articles: {
          selected: 1,
        },
      })).toEqual({
        selectedArticle: 1,
      });
    });
  });
});
