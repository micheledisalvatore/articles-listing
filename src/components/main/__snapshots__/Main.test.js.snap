// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a Main component when it is rendered should match the snapshot 1`] = `
<Fragment>
  <Connect(Grid) />
</Fragment>
`;
